src/webviews/extension-side/dataviewer/dataViewerDependencyService.ts

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import { SemVer } from 'semver';
+import { ProductNames } from '../../../kernels/installer/productNames';
+import { Product } from '../../../kernels/installer/types';
+import { IApplicationShell } from '../../../platform/common/application/types';
+import { traceWarning } from '../../../platform/logging';
+import { IsCodeSpace } from '../../../platform/common/types';
+import { parseSemVer } from '../../../platform/common/utils';
+import { DataScience, Common } from '../../../platform/common/utils/localize';
+import { EnvironmentType } from '../../../platform/pythonEnvironments/info';
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import { IDataViewerDependencyService, IDataViewerDependencyServiceOptions } from './types';
+import { executeSilently } from '../../../kernels/helpers';
+import { IKernel } from '../../../kernels/types';
+
+export const minimumSupportedPandaVersion = '0.20.0';
+export const kernelGetPandasVersion =
+    'import pandas as _VSCODE_pandas;print(_VSCODE_pandas.__version__);del _VSCODE_pandas';
+
+function isVersionOfPandaSupported(version: SemVer) {
+    return version.compare(minimumSupportedPandaVersion) > 0;
+}
+
+/**
+ * Responsible for managing dependencies of a Data Viewer.
+ */
+@injectable()
+export class DataViewerDependencyService implements IDataViewerDependencyService {
+    constructor(
+        @inject(IApplicationShell) private readonly applicationShell: IApplicationShell,
+        @inject(IsCodeSpace) private isCodeSpace: boolean
+    ) {}
+
+    private kernelPackaging(kernel: IKernel): '%conda' | '%pip' {
+        const envType = kernel.kernelConnectionMetadata.interpreter?.envType;
+        const isConda = envType === EnvironmentType.Conda;
+        console.log({ envType, isConda });
+        // From https://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-pip (%conda is here as well).
+        return isConda ? '%conda' : '%pip';
+    }
+
+    public async checkAndInstallMissingDependencies(options: IDataViewerDependencyServiceOptions): Promise<void> {
+        const kernel = options.kernel;
+        const interpreter = options.interpreter;
+
+        // Providing feedback as soon as possible.
+        if (!kernel && interpreter) {
+            sendTelemetryEvent(Telemetry.InterpreterNotSupportedOnTheWeb);
+            throw new Error(DataScience.interpreterNotSupportedOnTheWeb());
+        }
```

Comment:
Is it ok to throw if a developer provides an interpreter on the web and not a kernel? at least for now, while we work towards the debugger support.

Reply:
Yeah seems fine to me but this version is not web specific. It's the generic one that works for both. You'd throw this in a web specific one if you made one.

If your intent is to have one of these functions for node and one for web, I think you should create this method as abstract here and then override it in a 'web' class.